### PR TITLE
Add test coverage for unknown error types

### PR DESCRIPTION
**Pull request by jguidry-recurly**
Add a test case to verify that unknown error types are thrown as `ApiError`. I'm testing for the desired behavior [by way of `ArgumentException`](https://github.com/recurly/recurly-client-dotnet/blob/master/Recurly/Errors/Factory.cs#L189) which feels wonky. I'm open to any suggestions for a better way to work around [strict mode](https://github.com/recurly/recurly-client-dotnet/blob/master/Recurly/Utils.cs#L11) at test time to allow a more direct test.

See related [#640 in recurly-client-ruby](https://github.com/recurly/recurly-client-ruby/issues/640) and [#369 in recurly-client-python](https://github.com/recurly/recurly-client-python/issues/369) for context.

### CODE REVIEW

**Review by joannasese — APPROVED** [review]
This LGTM but I'd want another opinion on it before merging, perhaps from noted C# developer @douglasmiller

File: Recurly/BaseClient.cs
Comment (douglasmiller) [inline Recurly/BaseClient.cs:198]: 👍

**Review by douglasmiller — APPROVED** [review]
👍

**Pull request merged**
